graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -0,0 +1,342 @@
+
+This script allows us to run a PySpark job to perform the
+graph processing task, preparing the data for distributed
+partitioning. It is used as an entry point script with the
+`gs-processing` command.
+
+Usage:
+    distributed_executor.py --input-prefix <s3-or-local-prefix> \\
+        --output-prefix <s3-or-local-prefix> \\
+        --config-file <config-file>
+
+Script Parameters
+-----------------
+--input-prefix: str
+    S3 or local path prefix to the input data.
+--output-prefix: str
+    S3 or local path prefix to where the output data will be generated.
+--config-filename: str
+    GSProcessing configuration filename, needs to be under --input-prefix.
+--num-output-files: int
+    Number of output files Spark will create. Note that this affects
+    the available parallelism for the graph processing task, so do not
+    set this too low to ensure good performance. A good rule of thumb
+    is to set this to 2*num_of_cores_in_cluster. If not provided we
+    let Spark decide the parallelism level.
+--log-level: str
+    Log level for the script.
+--add-reverse-edges: str
+    When set to true (default), we add reverse edges for each edge type.
+
+"""
+import dataclasses
+import argparse
+import json
+import logging
+import os
+from pathlib import Path
+import time
+from typing import Any, Dict
+import tempfile
+
+import boto3
+
+from graphstorm_processing.graph_loaders.dist_heterogeneous_loader import (
+    DistHeterogeneousGraphLoader,
+)
+from graphstorm_processing.config.config_parser import create_config_objects
+from graphstorm_processing.config.config_conversion import GConstructConfigConverter
+from graphstorm_processing.data_transformations import spark_utils
+
+
+@dataclasses.dataclass
+class ExecutorConfig:
+    """Configuration  for the DistributedExecutor
+
+    Parameters
+    ----------
+    local_config_path : str
+        Local path to the config file.
+    local_output_path : str
+        Local path for output metadata files.
+    input_prefix : str
+        Prefix for input data. Can be S3 URI or local path.
+    output_prefix : str
+        Prefix for output data. Can be S3 URI or local path.
+    num_output_files : int
+        The number of output files Spark will try to create.
+    sm_execution : bool
+        Whether the execution context is a SageMaker container.
+    config_filename : str
+        The filename for the configuration file.
+    filesystem_type : str
+        The filesystem type, can be 'local' or 's3'.
+    add_reverse_edges : bool
+        Whether to create reverse edges for each edge type.
+    """
+
+    local_config_path: str
+    local_output_path: str
+    input_prefix: str
+    output_prefix: str
+    num_output_files: int
+    sm_execution: bool
+    config_filename: str
+    filesystem_type: str
+    add_reverse_edges: bool
+
+
+@dataclasses.dataclass
+class GSProcessingArguments:
+    """Arguments for the graph processing task"""
+
+    input_prefix: str
```

Comment:
Since these are replicating the arguments in parser_args and those have help text I thought I'd avoid duplicating it more. We already cover these a second time at the file-level docstring.